src/librustc_data_structures/tagged_ptr.rs

```diff
@@ -0,0 +1,125 @@
+//! This module implements tagged pointers.
+//!
+//! In order to utilize the pointer packing, you must have two types: a pointer,
+//! and a tag.
+//!
+//! The pointer must implement the `Pointer` trait, with the only requirement
+//! being conversion to and from a usize. The alignment of the Pointee dictates
+//! the space available in the least significant bits of the returned pointers.
+//!
+//! Note that we compare pointers by Pointee rather than the pointers
```

Comment:
This comment is confusing me. Compare for what? For the implementation packing/unpacking? Or just for `PartialEq`?

Also, by-value comparison is what we do for references and other smart pointers, so why is this even worth a comment? It's what I would have expected from a safe abstraction.